app/simulation_svc.py

```diff
@@ -98,3 +101,32 @@ def __init__(self, services, agents, loaded_scenario):
                 await self.start_agent(agent)
                 return True
         return False
+
+    """ PRIVATE"""
+
+    async def _extract_vars(self, ability, link):
+        decoded_test = self.decode_bytes(ability.test)
+        decoded_command = self.decode_bytes(link.command)
+        variables = re.findall(r'#{(.*?)}', decoded_test, flags=re.DOTALL)
+        vars = [x for x in variables if x not in ['server', 'paw', 'location', 'group']]
+        search_set = []
+        filtering = decoded_test
+        for v in vars:
+            search_key, filtering = filtering.split("#{" + v + "}", 1)
+            extract = re.findall(r'' + re.escape(search_key) + '(.*)', decoded_command, flags=re.DOTALL)
+            extract = extract[0]
```

Comment:
should be able to combine with the above:
`extract = (re.findall(r'' + re.escape(search_key) + '(.*)', decoded_command, flags=re.DOTALL))[0]`

Reply:
I could have sworn I put that back as one line (I broke it up for debugging at some point).